keps/sig-apps/4368-support-managed-by-label-for-batch-jobs/README.md

```diff
@@ -0,0 +1,924 @@
+recognize the new API, as in case of new fields. A similar approach was used
+to graduate the
+[Pod Index Label for StatefulSets and Indexed Jobs](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/4017-pod-index-label)
+feature with similar characteristics.
+
+Finally, we would like to have the feature soak without unnecessary delay.
+
+### Risks and Mitigations
+
+#### Ecosystem fragmentation due to forks
+
+The mechanism to disable the main Job controller opens the door for users to
+substitute it with a fork. This may create more fragmentation in the community
+as users may prefer to use their forked controllers rather than contribute
+upstream.
+
+First, this risk, to some extent, exists even today as admins with access to the
+control plane can disable by passing `--controllers=-job,*` in the manifest for
+`kube-controller-manager` (see more info on the `--controllers` flag
+[here](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/)).
+
+Second, we believe that users who had the need to fork the Job controller
+already introduced dedicated Job CRDs for their needs.
+
+#### Two controllers running at the same time on old version
+
+It is possible that one configures MultiKueue (or another project) with an
+older version of k8s which does not support the label yet. In that case
+two controllers might start running and compete with Job status updates at the
+same time.
+
+To mitigate this risk we warn about it in Kueue documentation, to use this label
+only against newer versions of Kubernetes.
+
+Finally, this risk will fade away with time as the new versions of
+Kubernetes support it.
+
+## Design Details
+
+#### API
+
+```golang
+const (
+  ...
+	// LabelManagedBy is used to indicate the controller or entity that manages
+	// an Job.
+	LabelManagedBy = "batch.kubernetes.io/managed-by"
+)
+```
+
+#### Implementation overview
+
+We skip synchronization of the Jobs with the `managed-by` label, if it has any
+different value than `job-controller.k8s.io`.
+
+There is no validation for the values of the field beyond that of standard
+permitted label values.
+
+#### Mutability
+
+We keep the label immutable for the first iteration of [Beta](#beta).
+Mutability could be handy for [Story 1](#story-1), but it is not a blocker.
+
+We will re-evaluate the need for mutability (and the actual conditions under
+which it is possible), before going [GA](#ga).
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+- `pkg/controller/job`: `2023-12-20` - `91.5%`
+- `pkg/registry/batch/job`: `2023-12-20` - `92.2%`
+- `pkg/apis/batch/v1`: `2023-12-20` - `29.3%` (mostly generated code)
+
+##### Integration tests
```

Comment:
Since we identified the immutability of the label, do you want to add an integration test case for that?

Reply:
Aren't unit tests enough in this case? For example, podFailurePolicy or podReplacementPolicy are immutable, but we don't have integration tests for it, just unit. 

For now, added only in the unit tests section. PTAL.

Reply:
Yes, I was thinking of ElasticJob feature where there are integration tests. But I think unit tests should be enough.

Reply:
Oh, I see, but there the logic is more complex, and the mutability is the heart of that feature. I'm fine either way, but let me keep as is for now.